### PR TITLE
[FIXED JENKINS-26777] Build reports to be running for 45 yr and counting

### PR DESCRIPTION
**Pull request by oldelvet**
Specifically check for the situation where the build has not started and return "Not started yet" in that case.

### CODE REVIEW

File: changelog.html [hunk 1]
```diff
@@ -56,6 +56,9 @@
 <div id="trunk" style="display:none"><!--=TRUNK-BEGIN=-->
 <ul class=image>
   <li class=>
+  <li class=bug>
```
Comment (daniel-beck) [inline changelog.html]: Should be added when merging, not be part of PR.

**Comment by daniel-beck** [comment]
:+1:

File: core/src/main/java/hudson/model/Run.java [hunk 2]
```diff
@@ -730,9 +730,12 @@ public String getDescription() {
      * Gets the string that says how long the build took to run.
      */
     public @Nonnull String getDurationString() {
-        if(isBuilding())
+        if (hasntStartedYet()) {
+            return Messages.Run_NotStartedYet();
+        } else if(isBuilding()) {
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Run.java]: space after if please

Comment (oldelvet) [inline core/src/main/java/hudson/model/Run.java]: Space fixed up. Thanks for the review.

File: core/src/main/java/hudson/model/Run.java [hunk 3]
```diff
@@ -730,9 +730,12 @@ public String getDescription() {
      * Gets the string that says how long the build took to run.
      */
     public @Nonnull String getDurationString() {
-        if(isBuilding())
+        if (hasntStartedYet()) {
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Run.java]: Interesting, why this method such named...

Comment (oldelvet) [inline core/src/main/java/hudson/model/Run.java]: I was not responsible for the name but "hasn't" it is a short form of "has not".

Comment (jglick) [inline core/src/main/java/hudson/model/Run.java]: It was a terrible choice of name (not just the misspelled contraction but the use of the negative sense), but too late now.

**Pull request merged**